mono/mini/memory-access.c

```diff
@@ -262,7 +262,7 @@ mini_emit_memory_copy_bytes (MonoCompile *cfg, MonoInst *dest, MonoInst *src, Mo
 		mini_emit_memory_barrier (cfg, MONO_MEMORY_BARRIER_SEQ);
 	}
 
-	if ((cfg->opt & MONO_OPT_INTRINS) && (size->opcode == OP_ICONST)) {
+	if ((cfg->opt & MONO_OPT_INTRINS) && (size->opcode == OP_ICONST) && size->inst_c0 < 10000) {
```

Comment:
I think it would be helpful to have a comment explaining how the number 10,000 was chosen.